### PR TITLE
sql: add an alias USE for SET DATABASE = ...

### PR DESCRIPTION
**Pull request by knz**
Prior to this patch, users that were interviewed trying to discover
CockroachDB proceeded as follows:

- if the user was coming from a MSSQL/MySQL background:
  - starts shell
  - runs "create database foo;"
  - runs "use foo;"
  - observes "syntax error"
  - is confused! (no clear doc pointer)

- if the user was coming from a postgres background:
  - starts shell
  - runs "create table"
  - observes: "no database specified"
  - briefly confused, and thinks "hmm I don't need to do this in
    Postgres. [pg has an implicit default db] Perhaps this is less
    postgres-y than I though, perhaps I should use the MySQL/MSSQL
    way to change databases?"
  - user tries to be smart and enters "create database foo; use foo;"
  - observes "syntax error"
  - is even more confused!

To summarize: non-pg users expect `USE` to work; pg users do not know
`set database =` because with a real pg server it is (most often) *not
needed*.

This patch thus introduces the syntax `USE <dbname>` (the MSSQL/MySQL
statement to "switch between" current databases) as an alias for `SET
database = <dbname>`.

### CODE REVIEW

**Comment by knz** [comment]
(note: I got this idea after attending some user interviews)

**Comment by jordanlewis** [comment]
+1 on idea alone - this confused me when I first joined as well.

**Review by madelynnblue — APPROVED** [review]

**Comment by knz** [comment]
TFYR
